### DISCUSSION TITLE
SOC_ROOT directory is not working if we mention it on application CMakeLists.txt

### DISCUSSION THREAD

**Discussion by aravinth-kalai** [zephyrproject-rtos/zephyr#67761] [Q&A] [answered]
Hello everyone!

I'm using zephyr 3.3.0 and micro controller as LPC55S16JBD64 , but in zephyr 3.3.0 there is no support for LPC55S16JBD64 (64 pin ) It's only having LPC55S16JBD100( 100 pin ) support, so I just edited the following soc folder files and now it's successfully  built
```
Kconfig.soc
Kconfig.defconfig.lpc55S16
soc.c 
```
Now the issue is I'm trying to keep the soc folder and files in my application folder and I changed the SOC_ROOT directory to my local application directory using main CMakeLists.txt , but if I build it it's still considering the zephyr stack's soc folder instead of my application directory soc folder , I verified this by reverting the changes in zephyr soc directory and keeping the changes in my application directory's soc, am I missing anything in my CMakeLists.txt

**rough application folder structure:**
```
main.c
CMakeLists.txt
soc\arm\nxp_lpc\lpc55xxx\CMakeLists.txt -> This soc folder is not considering for build
soc\arm\nxp_lpc\lpc55xxx\Kconfig.defconfig.lpc55S16
soc\arm\nxp_lpc\lpc55xxx\Kconfig.defconfig.series
soc\arm\nxp_lpc\lpc55xxx\Kconfig.series
soc\arm\nxp_lpc\lpc55xxx\Kconfig.soc
soc\arm\nxp_lpc\lpc55xxx\linker.ld
soc\arm\nxp_lpc\lpc55xxx\pinctrl_soc.h
soc\arm\nxp_lpc\lpc55xxx\soc.c
soc\arm\nxp_lpc\lpc55xxx\soc.h
soc\arm\nxp_lpc\CMakeLists.txt
soc\arm\CMakeLists.txt
soc\Kconfig
```

**zephyr 3.3.0 folder structure**
```
soc\arm\nxp_lpc\lpc55xxx\CMakeLists.txt -> This soc folder is consider for build
```

**main CMakeLists.txt**
```
set( SOC_ROOT   ${CMAKE_CURRENT_LIST_DIR} )
```
Please help me to fix this issue as soon as possible.

Thanks in advance !

**Comment by nordicjm** [comment]
> main CMakeLists.txt

Doesn't really show enough

**Reply by aravinth-kalai** [reply]
Hi @nordicjm ,

To change the soc directory we need to mention the below line

set( SOC_ROOT   ${CMAKE_CURRENT_LIST_DIR} )  in a main CMakeLists.txt list right?

Unfortunately I'm unable to share full CMakeLists.txt or If possible if you have any links related to this topic please share

Thank you !

**Comment by DerekSnell** [comment]
Hi @aravinth-kalai ,
When you set `SOC_ROOT` in your CMakeLists.txt, is that done before pulling in the Zephyr boilerplate with `find_package(Zephyr ...)`?

**Reply by aravinth-kalai** [reply]
Hi @DerekSnell ,

Yes SOC_ROOT is mentioned before
``` 
find_package( Zephyr REQUIRED HINTS $ENV{ZEPHYR_BASE} ) this line 

```

**Comment by DerekSnell** [comment, marked as answer]
Hi @aravinth-kalai ,
Is the issue a problem with missing Kconfig definitions?  Did you update all the paths sourced in the Kconfig files, to point to your app's SOC_ROOT folder?  For example, [lpc55xxx/Kconfig.defconfig.series](https://github.com/zephyrproject-rtos/zephyr/blob/zephyr-v3.3.0/soc/arm/nxp_lpc/lpc55xxx/Kconfig.defconfig.series) sources `soc/arm/nxp_lpc/lpc55xxx/Kconfig.defconfig.lp*`, which I believe points to the SOC folder in the Zephyr repo.  

A couple years ago I used my own SOC_ROOT folder for an SOC I added.  I struggled getting all my Kconfig files sourced.  I finally resolved it by putting intentional errors in my Kconfig files, and found some were not sourced.  Then I hunted down how each was getting source until I confirmed all of them were included in the build.

Let us know what you find.
